src/Callback.php

```diff
@@ -105,88 +94,88 @@ public function init(): void
         }
     }
 
     /**
      * Terminate this callback
-     * by rendering the owner view.
+     * by rendering the owner view by default.
      */
-    public function terminate()
+    public function terminateJson(View $view = null)
     {
         if ($this->canTerminate()) {
-            $this->app->terminateJson($this->owner);
+            $this->app->terminateJson($view ?: $this->owner);
         }
     }
 
     /**
-     * Prevent callback from terminating during a reload.
+     * Only current callback can terminate.
      */
-    protected function canTerminate(): bool
+    public function canTerminate(): bool
     {
-        $reload = $_GET['__atk_reload'] ?? null;
+        return $this->urlTrigger === ($_GET['__atk_callback'] ?? null);
+    }
 
-        return !$reload || $this->owner->name === $reload;
+    /**
+     * Allow callback to be triggered or not.
+     */
+    public function canTrigger(): bool
+    {
+        if ($this->triggerOnReload) {
+            return $this->isTriggered();
+        }
+
+        return $this->isTriggered() && !($_GET['__atk_reload'] ?? null);
+    }
+
+    /**
+     * Return true if urlTrigger is part of the request.
+     */
+    public function isTriggered()
+    {
+        return isset($_GET[$this->urlTrigger]);
     }
 
     /**
-     * Is callback triggered?
+     * Return callback triggered value.
      */
-    public function triggered()
+    public function getTriggeredValue(): string
     {
-        return $_GET[$this->urlTrigger] ?? false;
+        return $_GET[$this->urlTrigger] ?? '';
     }
 
     /**
      * Return URL that will trigger action on this call-back. If you intend to request
      * the URL direcly in your browser (as iframe, new tab, or document location), you
      * should use getUrl instead.
-     *
-     * @param string $mode
-     *
-     * @return string
      */
-    public function getJsUrl($mode = 'ajax')
+    public function getJsUrl(string $value = 'ajax'): string
     {
-        return $this->owner->jsUrl([$this->urlTrigger => $mode, '__atk_callback' => 1], (bool) $this->postTrigger);
+        return $this->owner->jsUrl($this->getUrlArguments($value));
     }
 
     /**
      * Return URL that will trigger action on this call-back. If you intend to request
      * the URL loading from inside JavaScript, it's always advised to use getJsUrl instead.
-     *
-     * @param string $mode
-     *
-     * @return string
      */
-    public function getUrl($mode = 'callback')
+    public function getUrl(string $value = 'callback'): string
+    {
+        return $this->owner->url($this->getUrlArguments($value));
+    }
+
+    /**
+     * Return proper url argument for this callback.
+     */
+    private function getUrlArguments(string $value): array
     {
-        return $this->owner->url([$this->urlTrigger => $mode, '__atk_callback' => 1], (bool) $this->postTrigger);
+        return ['__atk_callback' => $this->urlTrigger, $this->urlTrigger => $value];
```

Comment:
Setting value of `__atk_callback` implies last callback in the render tree, right?

It is used in `canTerminate()`. I think it will fail to reload callback containin another callback.